src/auth/auth.service.ts

```diff
@@ -0,0 +1,248 @@
+      },
+    );
+  }
+
+  async getTokens(id: string, email: string): Promise<Tokens> {
+    const [at, rt] = await Promise.all([
+      this.jwtService.signAsync(
+        { id, email },
+        {
+          secret: this.configService.get('SECRET_KEY_AT'),
+          expiresIn: this.configService.get('EXPIRES_IN_AT'),
+        },
+      ),
+      this.jwtService.signAsync(
+        { id, email },
+        {
+          secret: this.configService.get('SECRET_KEY_RT'),
+          expiresIn: this.configService.get('EXPIRES_IN_RT'),
+        },
+      ),
+    ]);
+    return { access_token: at, refresh_token: rt };
+  }
+
+  async checkUserByEmail(email: string): Promise<CheckUserResponse> {
+    const admin = await this.adminService.getAdminByEmail(email);
+
+    const hr = await this.hrService.getHrByEmail(email);
+
+    const student = await this.studentService.getStudentByEmail(email);
+
+    return hr ? hr : student ? student : admin ? admin : null;
+  }
+
+  async checkUserById(id: string): Promise<CheckUserResponse> {
+    const admin = await this.adminService.getUserById(id);
+
+    const hr = await this.hrService.getHrById(id);
+
+    const student = await this.studentService.getStudentById(id);
+
+    return admin ? admin : student ? student : hr ? hr : null;
+  }
+
+  async getUserData(user): Promise<UserDataResponse> {
+    const obj = {
+      id: user.id,
+      email: user.email,
+      role: user.role,
+    };
+    if (user instanceof Admin) {
+      return {
+        ...obj,
+        name: user.name,
+      };
+    }
+    if (user instanceof Hr) {
+      return {
+        ...obj,
+        fullName: user.fullName,
+      };
+    }
+    if (user instanceof Student) {
+      return {
+        ...obj,
+        firstName: user.firstName,
+        lastName: user.lastName,
+        githubUsername: user.githubUsername,
+      };
+    }
+  }
+
+  async login(login: LoginUserDto, response: Response): Promise<ApiResponse<UserDataResponse>> {
+    const user = await this.checkUserByEmail(login.email);
+    if (!user) throw new InvalidCredentialsException();
+
+    const passwordMatches = await this.compareHashedData(login.password, user.password);
+    if (!passwordMatches) throw new InvalidCredentialsException();
+
+    const tokens = await this.getTokens(user.id, user.email);
+    await this.updateRtHash(user.id, tokens.refresh_token);
+    response.cookie('jwt-refresh', tokens.refresh_token, { httpOnly: true });
+    return {
+      isSuccess: true,
+      payload: { ...(await this.getUserData(user)), access_token: tokens.access_token },
+    };
+  }
+
+  async logout(id: string, res: Response): Promise<ApiResponse<any>> {
+    const user = await this.checkUserById(id);
+    if (!user) throw new HttpException('Nie ma takiego użytkownika w systemie', HttpStatus.NOT_FOUND);
+    if (user.refreshToken !== null) {
+      user.refreshToken = null;
+      await user.save();
+      await this.cacheManager.del('filter');
```

Comment:
Mam niestety bardzo dużą obawę co do tego. Używacie jednego cache, które dotyczy wszystkich użytkowników, prawda? Czy to oznacza, że jak zalogowanych jest kilku HR-ów i jeden ustawi filtry, to ustawią się one u wszystkich?
Jest spora szansa, że tak właśnie jest - sprawdźcie to proszę, bo to jest bardzo poważny błąd.